src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -2461,27 +2461,51 @@ window.qBittorrent.DynamicTable ??= (() => {
             });
         }
 
-        #filterNodes(node, filterTerms, filteredRows) {
-            if (node.isFolder) {
-                const childAdded = node.children.reduce((acc, child) => {
-                    // we must execute the function before ORing w/ acc or we'll stop checking child nodes after the first successful match
-                    return (this.#filterNodes(child, filterTerms, filteredRows) || acc);
-                }, false);
+        #filterNodes(root, filterTerms) {
+            const ret = [];
+            const stack = [root];
+            const visited = [];
```

Comment:
Perhaps it would look clearer if you set `node.visited` property instead.

Reply:
Originally I didn't want to use property to avoid tainting the data. Now that point is moot since I already used `has_children_added`. I'll look into this.

Reply:
> Originally I didn't want to use property to avoid tainting the data.

I don't like it either. But seeing that it's already in use, I assumed that it's not considered "bad form" in JS. Ideally, I would just store a separate structure on the stack:
```c++
struct NodeTraverseContext
{
    Node *node;
    NodeTraverseContext *parent;
    bool visited;
    bool hasFilteredChildren;
    // whatever else needed
};
```

Reply:
>Ideally, I would just store a separate structure on the stack:

I'm concerned that it will need to allocate a new object and I'm not sure how much it would impact the performance.
I'll just stick to the current way since it is possible to remove the temporary properties when the loop finishes.